lib/plugins/crowdsec/metrics.lua

```diff
@@ -0,0 +1,192 @@
+-- @return the new value of the key
+function metrics:increment(key, increment, labels)
+    increment = increment or 1
+    if labels == nil then
+      ngx.log(ngx.INFO, "no labels")
+    end
+
+    -- keys could look like:
+    -- processed/ip_version=ipv4&
+    -- active_decisions/ip_version=ipv4&decision_type=ban&
+    key = key .. "/" .. utils.table_to_string(labels)
+    local value, err, forcible = self.cache:incr("metrics_" .. key, increment, 0)
+    metrics:add_to_metrics(key)
+    if err then
+        ngx.log(ngx.ERR, "failed to increment key: " .. key, err)
+    end
+    if forcible then
+      ngx.log(ngx.ERR, "Lua shared dict (crowdsec cache) is full, please increase dict size in config")
+    end
+    return value
+end
+
+-- Get all metrics as a table (key-value pairs)
+function metrics:get_all_keys()
+    local keys = metrics.cache:get("metrics_all")
+    return utils.split_on_delimiter(keys, ",")
+end
+
+-- Add a metric key to the `metrics_all` list
+function metrics:add_to_metrics(key)
+    local metrics_all = self.cache:get("metrics_all") or ""
+    if not metrics_all:find(key) then
+        metrics_all = metrics_all .. key .. ","
+        local success, err, forcible = self.cache:set("metrics_all", metrics_all)
+        if not success then
+          ngx.log(ngx.ERR, "failed to set key metrics_all: ", err)
+        end
+        if forcible then
+          ngx.log(ngx.ERR, "Lua shared dict (crowdsec cache) is full, please increase dict size in config")
+        end
+
+    end
+end
+
+--- Get the labels from a cache key
+--- As labels are stored in the cache key
+--- we need to extract them from the key
+--- @param key string key: the cache key to extract the labels from
+--- @return string the key without the labels and the labels as a table
+--- @return table labels as a table
+local function get_labels_from_key(key)
+  local table = utils.split_on_delimiter(key, "/")
+  local labels = {}
+  if table == nil then
+    return "", {}
+  else
+    if table[2] ~= nil then
+      labels = utils.string_to_table(table[2])
+    end
+  end
+  return table[1], labels
+end
+
+-- Export the store data to JSON
+function metrics:toJson(window)
+  local metrics_array = {}
+  local metrics_data = self.cache:get("metrics_data")
+  local keys = metrics:get_all_keys()
+  for  _,key in ipairs(keys) do
+    local cache_key = "metrics_" .. key
+    local value = self.cache:get(cache_key)
+    ngx.log(ngx.INFO, "cache_key: " .. cache_key .. " value: " .. tostring(self.cache:get(cache_key)))--debug
+    if value ~= nil then
+      ngx.log(ngx.INFO, "key: " .. key)
+      local final_key, labels = get_labels_from_key(key)
+      ngx.log(ngx.INFO, "final_key: " .. final_key)
+      ngx.log(ngx.INFO, "value: " .. value)
+      if labels ~= nil then
+        for k, v in pairs(labels) do
+          ngx.log(ngx.INFO, "label: " .. k .. " " .. v)
+        end
+      end
+
+      if final_key == "processed" then
```

Comment:
simplify code, we have only "active_decisions" that is IP related, the rest should be "request" related ?